### PR TITLE
Parse sdf's joint dynamics completely

### LINKED ISSUE #1382: SdfParser doesn't parse joint dynamics completely
**Is your feature request related to a problem? Please describe.**
Currently, SdfParser doesn't parse joint dynamics: friction, spring_reference, spring_stiffness.

### PR DESCRIPTION
**Pull request by vinnamkim**
- Parse joint friction, spring_stiffness, spring_reference

Signed-off-by: vinnamkim <vinnam.kim@gmail.com>

Resolve #1382 

***

**Before creating a pull request**

- [ ] Document new methods and classes
- [x] Format new code files using `clang-format`

**Before merging a pull request**

- [ ] Set version target by selecting a milestone on the right side
- [ ] Summarize this change in `CHANGELOG.md`
- [x] Add unit test(s) for this change

### CODE REVIEW

File: dart/utils/sdf/SdfParser.cpp [hunk 1]
```diff
@@ -1258,7 +1258,7 @@ static void readAxisElement(
     const Eigen::Isometry3d& _parentModelFrame,
     Eigen::Vector3d& axis,
     double& lower, double& upper, double& initial, double& rest,
-    double& damping)
+    double& damping, double &friction, double &sprint_stiffness)
```
Comment (jslee02) [inline dart/utils/sdf/SdfParser.cpp]:
```suggestion
    double& damping, double& friction, double& sprint_stiffness)
```

Comment (vinnamkim) [reply dart/utils/sdf/SdfParser.cpp]: Applied `clang-format` to `SdfParser.cpp` overall.

**Comment by vinnamkim** [comment]
> Thank you for the contribution! One style change is suggested in the inline comment.
> 
> Would you be able to add a simple unit test for completion something like [this](https://github.com/dartsim/dart/blob/master/unittests/unit/test_DartLoader.cpp#L98-L176) to [`test_SdfParser.cpp`](https://github.com/dartsim/dart/blob/master/unittests/unit/test_SdfParser.cpp)?

An error on parsing spring stiffness and related unit tests are added.

File: data/sdf/test/test_skeleton_joint.world [hunk 2]
```diff
@@ -0,0 +1,278 @@
+<?xml version="1.0" ?>
+<sdf version="1.4">
```
Comment (jslee02) [inline data/sdf/test/test_skeleton_joint.world]:
```suggestion
<sdf version="1.5">
```

Looking at the [specification](http://sdformat.org/spec?elem=joint&ver=1.5), the new joint dynamics properties were introduced in version 1.5.

Comment (vinnamkim) [reply data/sdf/test/test_skeleton_joint.world]: @jslee02 Resolved the conflicts and modified the sdf file version to 1.5

**Pull request merged**